src/saml2/sigver.py

```diff
@@ -211,6 +211,20 @@ def _get_xmlsec_cryptobackend(path=None, search_paths=None):
 ENC_KEY_CLASS = 'EncryptedKey'
 
 
+def get_environ_delete_tmpfiles():
+    default = "true"
+    value = os.environ.get("PYSAML2_DELETE_TMPFILES", default)
+    result = value.lower() == default
+
+    if not result:
+        logger.warning(
+            "PYSAML2_DELETE_TMPFILES set to False, "
+            "temporary xml files will not be deleted."
+        )
```

Comment:
I am not sure that this should be here. Ideally, this should show up once, not every time this function is invoked. The actual result of this function will not change, so, ideally the whole function should run only once.

Reply:
Actually It's executed two Times in the code but if you have another idea do as you prefer, the most important thing Is that now those temp files are removed. That log message should be executed only once, I agree

Reply:
@c00kiemon5ter is right. This should move to the Config reading part which is outside of this module. It doesn't really matter how many times, this is executed. We should try to have our configuration reading (and related warnings/logs etc) in one place. I'll see how we can refactor this part and post my suggestions here
PS: since adding to this PR I must first do a PR to @peppelinux branch, I was thinking of pasting the diff here (or stating the changes here) unless there's another suggestion for this discussion

Reply:
I agree this line. Go ahead, I'll do a commit with your contribution, referencing you in the commit description. Really thanks